Gems/Atom/RPI/Code/Source/RPI.Public/Image/ImageSystem.cpp

```diff
@@ -347,6 +389,14 @@ namespace AZ
                 Data::AssetId m_assetId;
             };
 
+            // Sync values from desc back to the cvars
+            // Note: we need this code because one of the instance of the cvar might be initialized early than setting registry
+            // due to this issue: https://github.com/o3de/o3de/issues/5537
```

Comment:
This issue has nothing to do with the Settings Registry and more to do with the CVar value being set in the data space of the loaded Gem modules at the time.
For example if the CVar is set when the `Atom_RPI.Editor` module is loaded, but before the `EMotionFX_Atom.Editor` module is loaded, then only the static variable copy of `r_streamingImagePoolBudgetMb` inside of `Atom_RPI.Editor` is set and not the static variable of `r_streamingImagePoolBudgetMb` within the `EMotionFX_Atom.Editor` module.

So performing a direct access of the `r_streamingImagePoolBudgetMb` could result in two different values being returned depending on which gem module the code is executing within.

Reply:
I added initialization function to set the default value from the loaded setting registry. So it shouldn’t need to set values here again and I intended to remove these lines here. But somehow when the first time the init function get called, the setting registry wasn’t created. So it won’t get the value.
My comment is for this use case but not for using the performcommand function vs directly assign

Reply:
Thank you for the clarification here.

Reply:
I wanted to use the GHI to explain why there might be multiple instances of the same cvar.  I may explain a bit more about the situation and remove the GHI since it may confuse people